### PR TITLE
Migrate to java 17

### PR DESCRIPTION
**Pull request by adituriya**
## Description
Migrate project to Java 17

## Motivation and Context
Java 8 deprecation

## How Has This Been Tested?
Ran tests locally and on CCI

## Release Plan:
- As the Submitter, upon requesting review of this pull request, I confirm that the [Release Checklist](https://help.risevision.com/hc/en-us/articles/360031119991) has been completed.
- As the Reviewer, upon approving the changes in this PR, I confirm I have reviewed and I agree that the [Release Checklist](https://help.risevision.com/hc/en-us/articles/360031119991) has been completed

#### Release Checklist Items Skipped?
No

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -16,33 +19,59 @@ jobs:
           paths:
             - ~/.m2
           key: client-side-events-{{ checksum "pom.xml" }}
-          
-  "deploy_prod":
+
+  deploy_stage:
```
Comment (fjvallarino) [inline .circleci/config.yml:23]: @adituriya is it safe to deploy to staging? It seems the target project is always `client-side-events`.

**Comment by alexey-rise** [comment]
@adituriya could you please squash all commits into one? Thanks.

File: .circleci/config.yml (hunk 1 above)
Comment (alex-deaconu) [reply .circleci/config.yml:23]: @fjvallarino this project doesn't really have a "staging" environment, so we decided to deploy to `client-side-events` and test the version manually. I'm not sure if we should revert that or keep it for future use. Thanks.

**Comment by alex-deaconu** [comment]
@alexey-rise I've squashed and rebased with master. Thanks.

**Comment by alex-deaconu** [comment]
To note this has been running in production since Thursday. Thanks.

**Review by alexey-rise — APPROVED** [review]
@adituriya @alex-deaconu looks good 👍 Thanks.

**Pull request merged**